crates/engine/tree/src/tree/mod.rs

```diff
@@ -1319,6 +1319,9 @@ where
             match self.insert_block(child) {
                 Ok(res) => {
                     debug!(target: "engine", child =?child_num_hash, ?res, "connected buffered block");
+                    if self.is_sync_target_head(child_num_hash.hash) {
+                        self.make_canonical(child_num_hash.hash);
+                    }
```

Comment:
this is the main change in the PR, without this the additional blocks after backfill sync were not being made canonical.

Reply:
oh very nice, yeah I actually noticed this and was wondering